cli/src/crypto/auth.rs

```diff
@@ -0,0 +1,130 @@
+use {
+    super::AuthArgs,
+    crate::{
+        command_title,
+        display::json_output,
+        loading,
+        prelude::*,
+        sui::{
+            build_sui_client,
+            create_wallet_context,
+            fetch_gas_coin,
+            fetch_reference_gas_price,
+            get_nexus_objects,
+            sign_and_execute_transaction,
+        },
+    },
+    anyhow::anyhow,
+    nexus_sdk::{
+        crypto::{session::Session, x3dh::IdentityKey},
+        idents::workflow,
+        object_crawler::fetch_one,
+        sui,
+        transactions::crypto::claim_pre_key_for_self,
+        types::Prekey,
+    },
+    serde_json::json,
+};
+
+pub(crate) async fn auth(args: AuthArgs) -> AnyResult<(), NexusCliError> {
```

Comment:
name this function something a bit more descriptive akin to the other modules please